client/components/advanced-credentials/index.tsx

```diff
@@ -89,26 +94,42 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 	const { protocol } = credentials;
 	const isAtomic = hasCredentials && 'dynamic-ssh' === protocol;
 
+	const credentialsTestStatus = useSelector( ( state ) =>
+		getJetpackCredentialsTestStatus( state, siteId, role )
+	);
+
 	const isRequestingCredentials = useSelector(
 		( state ) => isRequestingSiteCredentials( state, siteId as number ) || testCredentialsLoading
 	);
 
 	useEffect( () => {
-		try {
-			( async () => {
-				const results: { ok: boolean } = await wp.req.post( {
-					path: '/sites/' + siteId + '/rewind/credentials/test?role=main',
-					apiNamespace: 'wpcom/v2',
-				} );
-				const { ok } = results;
-				setTestCredentialsLoading( false );
-				setTestCredentialsResult( ok );
-			} )();
-		} catch ( error: any ) {
-			setTestCredentialsLoading( false );
-			setTestCredentialsResult( false );
+		if ( hasCredentials ) {
+			dispatch( testCredentials( siteId, role ) );
+		} else {
+			dispatch( markCredentialsAsInvalid( siteId, role ) );
 		}
-	}, [] );
+	}, [ hasCredentials ] );
+
+	useEffect(
+		function () {
+			if ( 'pending' !== credentialsTestStatus ) {
+				setTestCredentialsLoading( false );
+			}
+
+			if ( 'valid' === credentialsTestStatus ) {
+				setTestCredentialsResult( true );
+			}
+
+			if ( 'invalid' === credentialsTestStatus ) {
+				setTestCredentialsResult( false );
+			}
+
+			if ( 'failure' === credentialsTestStatus ) {
+				setTestCredentialsResult( false );
+			}
+		},
+		[ credentialsTestStatus ]
```

Comment:
Aren't `setTestCredentialsLoading` and `setTestCredentialsResult` also dependencies here?

Reply:
We are passing the `credentialsTestStatus` to this effect, as we want to re-run it when only the `credentialsTestStatus` changes. Is there any recommendation/suggestion about why it would be likely to include those as dependency for that effect (regardless of the implementation)?